framework/app.mk

```diff
@@ -81,8 +100,13 @@ app_deps     := $(patsubst %.$(obj-suffix), %.$(obj-suffix).d, $(objects)) \
                 $(patsubst %.C, %.$(obj-suffix).d, $(main_src)) \
                 $(ADDITIONAL_APP_DEPS)
 
+app_test_deps     := $(patsubst %.$(obj-suffix), %.$(obj-suffix).d, $(test_objects)) \
+                $(patsubst %.c, %.$(obj-suffix).d, $(test_csrcfiles))
 depend_dirs := $(foreach i, $(DEPEND_MODULES), $(MOOSE_DIR)/modules/$(i)/include)
 depend_dirs += $(APPLICATION_DIR)/include
+ifneq ($(wildcard $(APPLICATION_DIR)/test/include/*),)
```

Comment:
I don't think this would work for multilevel includes (why don't really have many, if any of these).